hash/hash.go

```diff
@@ -104,18 +90,14 @@ func (h Hash) Equal(h1 Hash) bool {
 	return bytes.Equal(h, h1)
 }
 
-// Marshal marshals hash into slice of base58 bytes. It's used by protobuf.
+// Marshal marshals hash into slice of bytes. It's used by protobuf.
```

Comment:
Are those function necessary? the type is still `[]byte`. maybe there is no need for it.

Reply:
Maybe, but I prefer to left them until I will be sure that they are not required by `customtype` extensions. If the are not then I will remove them :)